sdk/healthinsights/azure-health-insights-clinicalmatching/README.md

```diff
@@ -0,0 +1,139 @@
+
+#### Get API Key
+
+You can obtain the endpoint and API key from the resource information in the [Azure Portal][azure_portal].
+
+Alternatively, you can use the [Azure CLI][azure_cli] snippet below to get the API key from the Health Insights resource.
+
+```bash
+az cognitiveservices account keys list --resource-group <your-resource-group-name> --name <your-resource-name>
+```
+
+#### Create a ClinicalMatchingClient with an API Key Credential
+
+Once you have the value for the API key, you can pass it as a string into an instance of **AzureKeyCredential**. Use the key as the credential parameter
+to authenticate the client:
+
+```Java Snippet:
+String endpoint = "endpoint";
+String apiKey = "apiKey";
+ClinicalMatchingAsyncClient asyncClient = new ClinicalMatchingClientBuilder()
+    .endpoint(endpoint)
+    .serviceVersion(AzureHealthInsightsServiceVersion.getLatest())
+    .httpClient(HttpClient.createDefault(new HttpClientOptions()))
+    .credential(new AzureKeyCredential(apiKey))
+    .buildAsyncClient();
+
+```
+
+## Key concepts
+
+Trial Matcher provides the user of the services two main modes of operation: patients centric and clinical trial centric.
+- On patient centric mode, the Trial Matcher model bases the patient matching on the clinical condition, location, priorities, eligibility criteria, and other criteria that the patient and/or service users may choose to prioritize. The model helps narrow down and prioritize the set of relevant clinical trials to a smaller set of trials to start with, that the specific patient appears to be qualified for.
+- On clinical trial centric, the Trial Matcher is finding a group of patients potentially eligible to a clinical trial. The Trial Matcher narrows down the patients, first filtered on clinical condition and selected clinical observations, and then focuses on those patients who met the baseline criteria, to find the group of patients that appears to be eligible patients to a trial.
+
+## Examples
+
+Finding potential eligible trials for a patient.
+<!--
+- [SampleMatchTrialsSync.java](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/healthinsights/azure-health-insights-clinicalmatching/src/samples/java/com/azure/health/insights/clinicalmatching/SampleMatchTrialsSync.java).
+-->
+```Java readme-sample-MatchTrialsSync
+ClinicalMatchingClient syncClient = new ClinicalMatchingClientBuilder()
+    .endpoint(endpoint)
+    .serviceVersion(AzureHealthInsightsServiceVersion.getLatest())
+    .httpClient(HttpClient.createDefault(new HttpClientOptions()))
+    .credential(new AzureKeyCredential(apiKey))
+    .buildClient();
+
+SyncPoller<BinaryData, BinaryData> syncPoller = syncClient.beginMatchTrials(BinaryData.fromObject(trialMatcherOptions), null);
```

Comment:
I assume you'd like user to use `TrialMatcherData` for API. Then do this in sample (instead of `BinaryData`).

Reply:
readme updated